src/gui/MainPane/RoomList.qml

```diff
@@ -129,6 +129,32 @@ HListView {
         }
     }
 
+    // Find latest highlight or unread. If oldest=true, find oldest instead.
+    function latestUnreadRoom(oldest=false, highlights=false) {
+        const prop       = highlights ? "highlights" : "unreads"
+        const local_prop = highlights ? "highlights" : "local_unreads"
+
+        // When highlights=true, we don't actually find the latest highlight,
+        // but instead, the latest unread among all the highlighted rooms.
+
+        var max = null
+        var maxevent = null
```

Comment:
Use [`let`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/let#description) to declare modifiable local variables